### PR TITLE
fix(tools/respecDocWriter): emit all errors and warnings

### LINKED ISSUE #2062: Emit all errors/warning when with `-e -w`
Currently respec2html with `-e -w` halts when it meets the first error or warning message. It should instead emit all messages and then halt, so that the author can fix all things at once.

### PR DESCRIPTION
**Pull request by EvsChen**
Fix #2062

### CODE REVIEW

File: tools/respecDocWriter.js [hunk 1]
```diff
@@ -250,9 +263,6 @@ function makeConsoleMsgHandler(page) {
           console.warn(colors.warn(`🚨 ${output}`));
           break;
       }
-      if (abortOnError || abortOnWarning) {
```
Comment (marcoscaceres) [inline tools/respecDocWriter.js]: We still need to exit the process with the right code based on either a warning or error.

Comment (marcoscaceres) [reply tools/respecDocWriter.js]: Ah sorry, I see you move it to the top. Ignore this comment.

File: tools/respecDocWriter.js [hunk 2]
```diff
@@ -235,9 +244,13 @@ function makeConsoleMsgHandler(page) {
         // https://github.com/GoogleChrome/puppeteer/issues/1939
         return;
       }
-      const abortOnWarning = whenToHalt.haltOnWarn && type === "warning";
-      const abortOnError = whenToHalt.haltOnError && type === "error";
       const output = `ReSpec ${type}: ${colors.debug(text)}`;
+      if (type === "error") {
```
Comment (marcoscaceres) [inline tools/respecDocWriter.js]: Nit: use switch statement here I think.

File: tools/respecDocWriter.js [hunk 3]
```diff
@@ -91,6 +95,11 @@ async function fetchAndWrite(
     }
     await checkIfReSpec(page);
     const html = await generateHTML(page, url);
+    const abortOnWarning = whenToHalt.haltOnWarn && haltFlags.warn;
```
Comment (marcoscaceres) [inline tools/respecDocWriter.js]: I feel like the the `handleConsoleMessages()` should now move into the `generateHTML()` function, and that the `generateHTML()` should return the html AND the if it emitted an error and/or a warning.

File: tools/respecDocWriter.js (hunk 2 above)
Comment (saschanaz) [reply tools/respecDocWriter.js]: Maybe can be merged with the following switch.

File: tools/respecDocWriter.js (hunk 3 above)
Comment (saschanaz) [reply tools/respecDocWriter.js]: `handleConsoleMessages()` must be called before `page.goto()`, that was the point of #2057.

Comment (marcoscaceres) [reply tools/respecDocWriter.js]: Ah, keep forgetting. Sorry about that.

File: tools/respecDocWriter.js (hunk 2 above)
Comment (EvsChen) [reply tools/respecDocWriter.js]: Thanks for the advice! Already fixed.

**Comment by saschanaz** [comment]
I'm yet to locally test this, I'll do soon™️

**Pull request merged**